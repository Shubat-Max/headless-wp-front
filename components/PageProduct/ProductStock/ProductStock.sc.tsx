import styled from "styled-components";
import {Colors} from "../../../constants/colors";

export const StockItem = styled.div`
  background: #FAFAFA;
  border-radius: 3px;
  padding: 10px;
  display: flex;
  align-items: center;
  margin-top: 10px;
`
export const StockItemAttributes = styled.div`
  flex: 1;
`
export const StockItemAttribute = styled.div`
  font: normal normal 400 12px/1.5em 'Open-Sans', sans-serif;
  color: ${Colors.PRIMARY_TEXT};
  margin-top: 5px;
  
  &:first-of-type {
    margin: 0;
  }
`
export const StockItemShow = styled.div`
  flex: 0 0;
  border: 2px solid #C4C4C4;
  padding: 6px 13px;
  border-radius: 3px;
  font: normal normal 400 14px/1em 'Open-Sans', sans-serif;
  color: ${Colors.PRIMARY_TEXT};
  cursor: pointer;
  
  &:hover {
    border-color: ${Colors.PRIMARY}
  }
`


export const StockList = styled.div`
  padding-bottom: 20px;
  border-bottom: 1px solid ${Colors.DIVIDER};
`
export const Head = styled.div<any>`
  display:flex;
  align-items: center;
  cursor: pointer;
`
export const HeadDot = styled.div<{color}>`
  flex: 0 0 6px;
  height: 6px;
  width: 6px;
  border-radius: 50%;
  background: ${ props => props.color ? props.color : '#e3e3e3'};
  margin-right: 10px;
`
export const HeadTitle = styled.div`
  font: normal normal 400 14px/1.5em 'Open-Sans', sans-serif;
  color: ${Colors.PRIMARY_TEXT};
  flex: 1;
`
export const HeadArrow = styled.div<{active}>`
  flex: 0 0 10px;
  width: 10px;
  height: 10px;
  background: transparent url("data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAiIGhlaWdodD0iNiIgdmlld0JveD0iMCAwIDEwIDYiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CjxwYXRoIGQ9Ik05Ljg4NTg4IDAuNTg1ODk0QzkuNzMzNDggMC40MzMxNzkgOS40ODYxNyAwLjQzMjk0NCA5LjMzMzQ0IDAuNTg1MzI3TDUuMjc2MjEgNC42MzQxNEM1LjEyMzg5IDQuNzg2NDQgNC44NzYwOSA0Ljc4NjQyIDQuNzIzNSA0LjYzMzg0TDAuNjY2NTYzIDAuNTg1MzI3QzAuNTEzODQ4IDAuNDMyOTI1IDAuMjY2NTQzIDAuNDMzMTc5IDAuMTE0MTIxIDAuNTg1ODk0Qy0wLjAzODI2MTUgMC43Mzg2MDkgLTAuMDM4MDA3NiAwLjk4NTkzMyAwLjExNDY4OCAxLjEzODMyTDQuMTcxMzUgNS4xODY1NkM0LjM5OTg4IDUuNDE1MDUgNC42OTk5OCA1LjUyOTI3IDUuMDAwMTIgNS41MjkyN0M1LjMwMDA4IDUuNTI5MjcgNS42MDAxIDUuNDE1MDkgNS44MjgzNCA1LjE4NjgzTDkuODg1MzEgMS4xMzgzMkMxMC4wMzggMC45ODU5MzMgMTAuMDM4MyAwLjczODYwOSA5Ljg4NTg4IDAuNTg1ODk0WiIgZmlsbD0iIzM1MzIzOCIvPgo8L3N2Zz4K") center/10px no-repeat;
  transform: rotateZ( ${ props => props.active ? '-180deg' : '0deg'} );
  transition: .1s;
`
export const List = styled.div<{active}>`
  height: ${props => props.active ? 'auto': '0px'};
  overflow: hidden;
`
export const Tab = styled.div`
  margin-bottom: 10px;
  
  &:last-of-type {
    margin-bottom: 0;
  }
`