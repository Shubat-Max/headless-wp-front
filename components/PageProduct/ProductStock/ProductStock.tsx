import * as React from 'react';
import {useSelector} from "react-redux";
import {RootState} from "../../../redux/store";
import {
  Head,
  HeadArrow,
  HeadDot,
  HeadTitle, List, StockItem,
  StockItemAttribute,
  StockItemAttributes,
  StockItemShow,
  StockList,
  Tab
} from "./ProductStock.sc";

const mock = {
  stock: [{
    attrs: [
      'Цвет: серый',
      'Размер корпуса: 12х32х67',
      'Местоположение: ул. Магнитогорская, д. 30'
    ]
  },{
    attrs: [
      'Цвет: буро-малиновый',
      'Размер корпуса: 12х32х67',
      'Местоположение: ул. Магнитогорская, д. 30'
    ]
  },{
    attrs: [
      'Цвет: красный',
      'Размер корпуса: 12х32х67',
      'Местоположение: ул. Магнитогорская, д. 30'
    ]
  }],
  expo: [{
    attrs: [
      'Цвет: буро-малиновый',
      'Размер корпуса: 12х32х67',
      'Местоположение: ул. Магнитогорская, д. 30'
    ]
  },{
    attrs: [
      'Цвет: красный',
      'Размер корпуса: 12х32х67',
      'Местоположение: ул. Магнитогорская, д. 30'
    ]
  }],
  outlet: [{
    attrs: [
      'Цвет: красный',
      'Размер корпуса: 12х32х67',
      'Местоположение: ул. Магнитогорская, д. 30'
    ]
  }]
}

const ProductStock = () => {

  const available = useSelector((state:RootState) => state.product.entity?.available )
  // const available = mock

  if( ! available ) return null
  // if( ! available ) return <StockList style={{color: '#e3e3e3'}}>Отсутствуют на складах</StockList>

  const [activeTab, setActiveTab] = React.useState( null )

  return (
    <StockList>

      { Object.keys(available).map((key, o) => {

        if( ! mock[key].length ) return null;

        let dotColor = '#e3e3e3';
        let title = 'Представленные товары'
        switch( key ){
          case 'stock': {
            dotColor = '#84E296'; title = 'В наличии'; break;
          }
          case 'expo': {
            dotColor = '#FFE55F'; title = 'На экспозиции'; break;
          }
          case 'outlet': {
            dotColor = '#EB5E55'; title = 'OUTLET'; break;
          }
        }

        const isTabActive = activeTab === key;

        return (
          <Tab key={`t-${o}`}>
            <Head onClick={() => { activeTab === key ? setActiveTab(null) : setActiveTab(key)}}>
              <HeadDot color={dotColor} />
              <HeadTitle>{title} ({mock[key].length})</HeadTitle>
              <HeadArrow active={isTabActive} />
            </Head>
            <List active={isTabActive}>
              {mock[key].map((item, i) =>
                <StockItem key={`i-${i}`}>
                  <StockItemAttributes>
                    {
                      item.attrs ? item.attrs.map((attr, j) => <StockItemAttribute key={`a-${j}`}>{attr}</StockItemAttribute>) : null
                    }
                  </StockItemAttributes>
                  <StockItemShow onClick={() => { console.log('@@ACTION/LOAD_STOCK') }}>Посмотреть</StockItemShow>
                </StockItem>
              )}
            </List>
          </Tab>
        )
      })}

    </StockList>
  );
};



export default ProductStock;