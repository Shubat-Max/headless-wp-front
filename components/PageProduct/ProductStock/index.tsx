import ProductStock from "./ProductStock";

export default ProductStock;