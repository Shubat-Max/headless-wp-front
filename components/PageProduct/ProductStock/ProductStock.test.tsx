import * as React from "react"
import * as redux from 'react-redux'
import {mount} from "enzyme"
import ProductStock from "./ProductStock";
import {StockList} from "./ProductStock.sc";
import renderer from 'react-test-renderer';

const spy = jest.spyOn(redux, 'useSelector')

describe('component', () => {
  describe('when product entity not avialable', () => {
    beforeEach(() => {
      spy.mockReturnValue(undefined)
    })

    it('should NOT render <StockList>', () => {
      const component = mount(<ProductStock/>);
      expect(component.find(StockList).exists()).toBeFalsy()
    })
  })
})