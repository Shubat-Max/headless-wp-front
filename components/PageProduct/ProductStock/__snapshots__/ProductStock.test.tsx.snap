// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component when product renderer <ProductStock> should NOT render <StockList> 1`] = `ReactWrapper {}`;
